core/trino-main/src/main/java/io/trino/dispatcher/DispatchManager.java

```diff
@@ -240,7 +248,7 @@ private <C> void createQueryInternal(QueryId queryId, Span querySpan, Slug slug,
             DispatchQuery dispatchQuery = dispatchQueryFactory.createDispatchQuery(
```

Comment:
this automatically also handles things like event listener and `QueryResource` right?

Might be worth to explicitly call it out in the commit message (although you do imply that by mentioning anything using QueryInfo/BasicQueryInfo).